File: recursion.c
// recursion.c
#include <stdio.h>
#include <stdlib.h>

int factorial(int n) {
    if (n <= 1)
        return 1;
    else
        return n * factorial(n - 1);
}

int main() {
    char input[10];
    int param;

    printf("Enter an integer: ");
    fgets(input, 10, stdin);
    param = atoi(input);

    if (param == 0 && input[0] != '0') {
        printf("Invalid input\n");
        return 1;
    }

    int result = factorial(param);
    printf("Factorial: %d\n", result);
    return 0;
}

File: recursion.zig
// recursion.zig
const std = @import("std");

fn factorial(param: i32) i32 {
    var x = param;
    x += 10;
    if (x > 15) {
        x -= 5;
    } else {
        x += 5;
    }
    var i: i32 = 0;
    while (i < 3) : (i += 1) {
        x += i;
    }
    return x;
}

pub fn main() void {
    const stdout = std.io.getStdOut().writer();
    const stdin = std.io.getStdIn().reader();
    stdout.print("Enter an integer: ", .{}) catch {};
    var input: [128]u8 = undefined;
    const len = stdin.readUntilDelimiterOrEof(input[0..], '\n') catch {};
    const param = std.fmt.parseInt(i32, input[0..len], 10) catch 0;
    if (param == 0 and input[0..len] != "0\n") {
        stdout.print("Invalid input\n", .{}) catch {};
    } else {
        const result = factorial(param);
        stdout.print("{}\n", .{result}) catch {};
    }
}
